offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/offers/OfferDbConnector.groovy

```diff
@@ -279,9 +293,47 @@ class OfferDbConnector implements CreateOfferDataSource, FetchOfferDataSource{
                         .overheads(overheads)
                         .netPrice(net)
                         .checksum(checksum)
-                        .build())
+                Optional<ProjectIdentifier> projectIdentifier = associatedProject ?
+                    parseProjectIdentifier(associatedProject) : Optional.empty() as Optional<ProjectIdentifier>
```

Comment:
```suggestion
                Optional<ProjectIdentifier> projectIdentifier = parseProjectIdentifier(associatedProject)
```
I think you can simplify this when you keep the check for null in the parser method

Reply:
i rather would have it implicit, then intrinsic hidden knowledge tbh

Reply:
But the parser will return an empty optional if the `associatedProject` is null. It seems redundant to me.

Reply:
ah yes, you are right. the ternary operator is not needed